hub/core/chunk_engine/tests/common.py

```diff
@@ -0,0 +1,154 @@
+import numpy as np
+import pickle
+
+from hub.core.chunk_engine import write_array, read_array
+from hub.core.chunk_engine.util import (
+    normalize_and_batchify_shape,
+    get_meta_key,
+    get_index_map_key,
+    get_chunk_key,
+    get_random_array,
+)
+from hub.core.storage import MappedProvider
+from hub.core.typing import Provider
+
+from typing import List, Tuple
+
+
+TENSOR_KEY = "TEST_TENSOR"
+
+
+STORAGE_PROVIDERS = (
+    MappedProvider(),
+)  # TODO: replace MappedProvider with MemoryProvider
+
+
+CHUNK_SIZES = (
+    128,
+    4096,
+    16000000,  # 16MB
+)
+
+
+DTYPES = (
+    "uint8",
+    "int64",
+    "float64",
+    "bool",
+)
+
+
+def get_min_shape(batch: np.ndarray) -> Tuple:
+    return tuple(np.minimum.reduce([sample.shape for sample in batch]))
+
+
+def get_max_shape(batch: np.ndarray) -> Tuple:
+    return tuple(np.maximum.reduce([sample.shape for sample in batch]))
+
+
+def assert_meta_is_valid(meta: dict, expected_meta: dict):
+    for k, v in expected_meta.items():
+        assert k in meta
+        assert v == meta[k]
+
+
+def assert_chunk_sizes(key: str, index_map: List, chunk_size: int, storage: Provider):
+    incomplete_chunk_names = set()
+    complete_chunk_count = 0
+    total_chunks = 0
+    for i, entry in enumerate(index_map):
+        for j, chunk_name in enumerate(entry["chunk_names"]):
+            chunk_key = get_chunk_key(key, chunk_name)
+            chunk_length = len(storage[chunk_key])
+
+            # exceeding chunk_size is never acceptable
+            assert (
```

Comment:
Codecov is complaining that this line is never hit by tests. Is this bad reporting? Assertions feel like the first thing that should be covered by tests

Reply:
yeah i noticed a lot of lines were being flagged but that were definitely covered. a lot of my tests had been flagged. which is funny cus i remember seeing like 10 flags inside the same test